datasets/signalmixer.py

```diff
@@ -28,109 +41,111 @@ class signalmixer:
-            signal_dataset, batch_size=signal_dataset.data.shape[0])))[0]
+        self.snr_range = snr_range
 
-        if noise_kind == 'WhiteNoise':
-            self.mixed_signal = self.white_noise_mixer()
+        if apply_prob > 1:
+            print('Noise will be applied to the whole dataset! \n')
+            self.apply_prob = 1
         else:
-            self.mixed_signal = self.snr_mixer()
+            self.apply_prob = apply_prob
+        self.noise_type = noise_type
 
     def __getitem__(self, index):
+        inp, target = self.signal_dataset[index]
+        data_type = self.data_type[index]
 
-        inp = self.mixed_signal[index].type(torch.FloatTensor)
-        target = int(self.signal_targets[index])
-        return inp, target
+        # applying noise to train set using per-snr
+        if data_type == 0:
+            if np.random.uniform(0, 1) < self.apply_prob:
+                if self.noise_type is not None:
+                    if not torch.all(inp == 0):
+                        inp = self.single_snr_mixer(inp)
+
+        # applying noise to all of the test set
+        else:
+            if 'WhiteNoise' in self.noise_type:
+                if not torch.all(inp == 0):
+                    inp = self.white_noise_mixer(inp)
+            elif self.noise_type is not None:
+                if not torch.all(inp == 0):
+                    inp = self.single_snr_mixer(inp)
+        return inp.type(torch.FloatTensor), int(target)
 
     def __len__(self):
-        return len(self.mixed_signal)
+        return len(self.data)
 
-    def snr_mixer(self):
-        ''' creates mixed signal dataset using the SNR level and the noise dataset
+    def single_snr_mixer(self, signal):
         '''
-        clean = self.test_dataset_float
-        noise = self.noise_dataset_float
-
-        idx = np.random.randint(0, noise.shape[0], clean.shape[0])
-        noise = noise[idx]
-        rms_noise = self.noise_rms[idx]
+        Creates mixed signals using an SNR level and the noise dataset
+        '''
+        idx_noise = np.random.randint(0, len(self.data))
+        noise, _ = self.noise_dataset[idx_noise]
+        rms_noise = self.noise_dataset.rms[idx_noise]
 
-        snr = self.snr
+        snr = self.snr_range
+        if snr[0] == snr[-1]:
+            snr_val = snr[0]
+        else:
+            snr_val = np.random.randint(snr[0], snr[-1])
+        snr_val = torch.tensor(snr_val)
 
-        rmsclean = torch.sqrt(torch.mean(clean.reshape(
-            clean.shape[0], -1)**2, 1, keepdims=True)).unsqueeze(1)
+        rmsclean = torch.sqrt(torch.mean(signal**2))
         scalarclean = 1 / rmsclean
-        clean = clean * scalarclean
+        signal = signal * scalarclean
 
-        scalarnoise = 1 / rms_noise.reshape(-1, 1, 1)
+        scalarnoise = 1 / rms_noise
         noise = noise * scalarnoise
 
-        cleanfactor = 10**(snr/20)
-        noisyspeech = cleanfactor * clean + noise
+        cleanfactor = 10**(snr_val/20)
```

Comment:
For better readability, consider keeping the `self.snr_range` variable (don't assign it to `snr`), and replacing `snr_val` with `snr`.